src/app/frontend/deploy/deployfromsettings.html

```diff
@@ -17,17 +17,19 @@
 <kd-help-section>
   <md-input-container class="md-block" md-is-error="ctrl.isNameError()">
     <label>App name</label>
-    <input ng-model="ctrl.name" name="name" namespace="ctrl.namespace" required kd-unique-name
-        ng-model-options="{ updateOn: 'default blur', debounce: { 'default': 500, 'blur': 0 } }">
-    <md-progress-linear class="kd-deploy-form-progress" md-mode="indeterminate"
-        ng-show="ctrl.form.name.$pending">
-    </md-progress-linear>
-    <ng-messages for="ctrl.form.name.$error" role="alert" multiple>
-      <ng-message when="required">Application name is required.</ng-message>
-      <ng-message when="uniqueName">
-        Application with this name already exists within namespace <i>{{ctrl.namespace}}</i>.
-      </ng-message>
-    </ng-messages>
+    <input ng-model="ctrl.name"
+           name="name"
+           namespace="ctrl.namespace"
+           required
+           kd-unique-name
+           ng-pattern="/^[a-z]+[a-z\-]*[a-z]*$/">
+
+    <div ng-messages="ctrl.form.name.$error" ng-if="ctrl.form.name.$touched">
```

Comment:
> BTW: I don't understand the purpose of md-is-error="ctrl.isNameError()". I checked documentation and the test cases, but could not make any clue out of it. For me it seems buggy and superfluous, but probably I just lack understanding...

Normally errors are shown only when focusing-out. For names we show errors at the time of typing. This is because name validation is async. Does this make sense?